cnd/src/network.rs

```diff
@@ -1131,6 +1137,185 @@ impl libp2p::swarm::NetworkBehaviourEventProcess<()> for ComitNode {
+                }
+
+                if role == Role::Alice {
+                    tokio::task::spawn({
+                        let connector = self.ethereum_connector.clone();
+                        let alice_ethereum_identity = create_swap_params.ethereum_identity;
+                        let bob_ethereum_identity = ethereum_identity;
+
+                        let asset = create_swap_params.ethereum_amount.clone();
+                        let ledger = ledger::Ethereum::default();
+                        let expiry = create_swap_params.ethereum_absolute_expiry;
+                        let secret_hash = secret_hash;
+
+                        new_han_ethereum_ether_swap(
+                            local_swap_id,
+                            connector,
+                            ethereum_ledger_state,
+                            HtlcParams {
+                                asset,
+                                ledger,
+                                redeem_identity: bob_ethereum_identity,
+                                refund_identity: alice_ethereum_identity.into(),
+                                expiry,
+                                secret_hash,
+                            },
+                            role,
+                        )
+                    });
+                } else {
+                    tokio::task::spawn({
+                        // This is Bob
+                        let connector = self.ethereum_connector.clone();
+                        let alice_ethereum_identity = ethereum_identity;
+                        let bob_ethereum_identity = create_swap_params.ethereum_identity;
+
+                        let asset = create_swap_params.ethereum_amount.clone();
+                        let ledger = ledger::Ethereum::default();
+                        let expiry = create_swap_params.ethereum_absolute_expiry;
+                        let secret_hash = secret_hash;
+
+                        new_han_ethereum_ether_swap(
+                            local_swap_id,
+                            connector,
+                            ethereum_ledger_state,
+                            HtlcParams {
+                                asset,
+                                ledger,
+                                redeem_identity: bob_ethereum_identity.into(),
+                                refund_identity: alice_ethereum_identity,
+                                expiry,
+                                secret_hash,
+                            },
+                            role,
+                        )
+                    });
+                }
+            }
+        }
+    }
+}
+
+/// Creates a new instance of the halight protocol.
+///
+/// This function delegates to the `halight` module for the actual protocol
+/// implementation. Its main purpose is to annotate the protocol instance with
+/// logging information and store the events yielded by the protocol in
+/// `InvoiceStates`.
```

Comment:
```suggestion
/// `halight::States`.
```
This is my bad :)

Reply:
note: I did a bit more cleanup to make the naming more consistent